app/controllers/config_reports_controller.rb

```diff
@@ -1,10 +1,20 @@
 class ConfigReportsController < ApplicationController
   include Foreman::Controller::AutoCompleteSearch
+  include Foreman::Controller::CsvResponder
 
   before_action :setup_search_options, :only => :index
 
   def index
-    @config_reports = resource_base_search_and_page(:host)
+    respond_to do |format|
+      format.html do
+        @config_reports = resource_base_search_and_page(:host)
+        render :index
+      end
+      format.csv do
+        @config_reports = resource_base_with_search.preload(:host)
```

Comment:
it actually tests the fact the server response and go though this controller ? this will catch any issues with the columns or other customization?